### PR TITLE
(Re)Add Leaderboards

### PR DESCRIPTION
**Pull request by AlbanTrincherini**
This PR does the following:
* Refactor the existing Leaderboard class to allow the creation of TeamLeaderboard and UserLeadeboard.
* Add a new PointRepository to allow the creation of the UserLeaderboard, this repository stores total points of each user.
* Add ViewModels for both the UserLeaderboard and TeamLeaderboard.

There were also some small bugs I encountered while working on this and fixed:
* Claiming a hunt now correctly removes the hunt from the ActiveHunts screen.
* The ProfilePage now correctly fetches the user's claims.

This PR closes user story #201, and issues #214 #215 #216

### CODE REVIEW

File: app/src/main/java/com/github/geohunt/app/ui/screens/userleaderboard/UserLeaderboardViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        userRepository: UserRepositoryInterface
+): ViewModel() {
+
+    private val _leaderboardInformation = MutableStateFlow(LeaderboardInformation(listOf(), -1))
+    val leaderboardInformation = _leaderboardInformation.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            val currentUser = userRepository.getCurrentUser().id
+            val topUsers = scoreRepository.getTopNUsers(N)
```
[the remark this answers is not in the document]
Comment (Maeeen) [reply app/src/main/java/com/github/geohunt/app/ui/screens/userleaderboard/UserLeaderboardViewModel.kt:36]: Consider changing it to something like "nb_fetch_users" or similar.

File: app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/Leaderboard.kt [hunk 2]
```diff
@@ -34,15 +36,15 @@ fun Leaderboard(
...
+        LeaderboardList(entries = entries, Modifier.weight(1.0f))
 
         // Bottom "You" item
-        LeaderboardListItem(
-            position = users.indexOf(currentUser),
-            user = currentUser,
-            isCurrent = true,
-            score = 0 // TODO
-        )
+        if(currentIndex != -1) {
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/Leaderboard.kt:42]: if currentIndex ≥ 0 is better?

File: app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/LeaderboardListItem.kt [hunk 3]
```diff
@@ -54,15 +51,15 @@ fun LeaderboardListItem(
             color = textColor
         )
 
-        ProfileIcon(user = user)
+        entry.displayIcon?.invoke()
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/LeaderboardListItem.kt:54]: Why invoke when you can just do `entry.displayIcon?()` ?

**Review by Maeeen — APPROVED** [review]
Looks very good to me. Check the really few minor comments, best

File: app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/LeaderboardListItem.kt (hunk 3 above)
Comment (AlbanTrincherini) [reply app/src/main/java/com/github/geohunt/app/ui/components/leaderboard/LeaderboardListItem.kt:54]: This sadly doesn't work 😢 😭 
You can't use something else than a method after ?

File: app/src/main/java/com/github/geohunt/app/ui/screens/userleaderboard/UserLeaderboardViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlbanTrincherini) [reply app/src/main/java/com/github/geohunt/app/ui/screens/userleaderboard/UserLeaderboardViewModel.kt:36]: Really love the name you proposed, will be changing it ASAP !!! 👍 💯

**Review by Maeeen — APPROVED** [review]

**Pull request merged**